modules/request.py

```diff
@@ -666,3 +669,35 @@ def close_log_files(self) -> None:
         for measurement in self._get_measurements():
             measurement.close_log_files()
         RequestLogger.close_log_files(self)
+
+    def import_external_file(self, imported_file_path: Path) -> Path or bool:
+        file_in_path = \
+            self.get_imported_files_folder() / imported_file_path.name
+
+        if file_in_path.exists():
+            try:
+                return False
+            except ValueError:
+                raise "The imported external file %s with the same name " \
+                      "has already been imported" % imported_file_path.name
```

Comment:
* `return False` can't throw a `ValueError` 
   * So what do you suggest?

 *  a `str` can't be `raise`d 
    * So what do you suggest?
   
*  f-strings (`f"hello {variable} world"`) are the main way to format strings in Potku 
    * I can change it, but why is that / does it matter? Placeholder are used too.

Reply:
> * `return False` can't throw a `ValueError`
>       
>       * So what do you suggest?

Looks like you were trying to check if the newly-imported file's name collides with another imported file's name. Do that check I guess.

>     * a `str` can't be `raise`d
>       
>       * So what do you suggest?

Raise a `ValueError` or something similar.

>     * f-strings (`f"hello {variable} world"`) are the main way to format strings in Potku
>       
>       * I can change it, but why is that / does it matter? Placeholder are used too.

f-strings are better: they have named substitutions (useful when formatting a long string) and they handle calling `str` automatically. For a more in-depth description, see [here](https://realpython.com/python-f-strings/).

The %-formatting and `.format()` are holdovers from old Python versions: f-strings were added in Python 3.6, so they couldn't have been used before that. It's a similar thing to `str` file paths, as those more or less replaced with `pathlib.Path`.

Reply:
There were some mistakes in the modified tests, the [originals](https://github.com/JYU-IBA/potku/pull/239#issuecomment-1014757126) seem fine. `self.assertFalse()` shouldn't be called for the tested `Request` methods, only for test stuff like `self.assertFalse(imported_file.exists())`.

A fixed test (there are others that still need to be fixed):
```python
    def test_error_is_raised_if_file_with_same_name_has_already_been_imported(
            self):
        with tempfile.TemporaryDirectory() as tmp_dir:
            tmp_dir = Path(tmp_dir)
            request = Request(directory=tmp_dir, **self.request_kwargs)
            external_file = self.create_external_file(tmp_dir)

            request.import_external_file(external_file)
            self.assertRaises(
                ValueError, lambda: request.import_external_file(external_file)
            )
```

Fixed `Request.import_external_file()` (do a similar thing for `remove_external_file`):
```python
    def import_external_file(self, imported_file_path: Path) -> Path:
        file_in_path = \
            self.get_imported_files_folder() / imported_file_path.name

        if file_in_path.exists():
            raise ValueError(
                f"Another external file '{imported_file_path.name}'"
                f" with the same name has already been imported")

        if not file_in_path.parent.exists():
            file_in_path.parent.mkdir(parents=True)

        shutil.copyfile(imported_file_path, file_in_path)

        return file_in_path
```

Changes:
- fixed return type and its annotation. @jkpnen by the way, `Path or bool` resolves to just `Path`. Use `from typing import Union; Union[Path, bool]` for that (or don't, because a single function/method shouldn't return different types).
- changed legacy `os` functions to `Path` methods
- updated error message

Fixed `EnergySpectrumParamsDialog._import_external_file()` (do a similar thing for `_remove_external_file`):

```python
    def _import_external_file(self):
        imported_external_files_paths = fdialogs.open_files_dialog(
            self, self.element_simulation.request.directory,
            "Select a file to import", "*.hist *.simu")
        if not imported_external_files_paths:
            return
        else:
            for file_path in imported_external_files_paths:
                try:
                    self.element_simulation.request.import_external_file(
                        file_path)
                except ValueError as e:
                    QtWidgets.QMessageBox.critical(
                        self, "Error", str(e),
                        QtWidgets.QMessageBox.Ok, QtWidgets.QMessageBox.Ok)
            self.external_tree_widget.clear()
            self._set_external_files()
```

Changes:
- added try-except